src/jenkins.py

```diff
@@ -297,25 +311,24 @@ def _install_plugins(
         JenkinsPluginError: if an error occurred installing the plugin.
     """
     command = ["java"]
-    if proxy_config:
+    if proxy_config and proxy_config.http_proxy:
         command += [
-            f"-Dhttp.proxyHost={proxy_config.hostname}",
-            f"-Dhttp.proxyPort={proxy_config.port}",
-            f"-Dhttps.proxyHost={proxy_config.hostname}",
-            f"-Dhttps.proxyPort={proxy_config.port}",
+            f"-Dhttp.proxyHost={proxy_config.http_proxy.host}",
+            f"-Dhttp.proxyPort={proxy_config.http_proxy.port}",
+            f"-Dhttp.proxyUser={proxy_config.http_proxy.user or ''}",
```

Comment:
This looks a bit like what we did above? Any opportunities to consolidate, especially the user and password bit?

Reply:
http_proxy is of type `HttpUrl` given by pydantic, so by default when user/password isn't given, it will be of type `None`. What I could suggest is to implement another method on `ProxyConfig` class to output a `typedDict` or a `NamedTuple` that can return a default value of empty string instead of `None`. Would this be better?

Reply:
I guess try it and we'll see 🙂

Reply:
refactored, thank you!